test/test_parser.py

```diff
@@ -0,0 +1,231 @@
+            .add_token(TokenTypes.FIELD)\
+            .add_key_value_pair(key, value)
+
+        parsed_key, parsed_value = Parser(lexer).field()
+
+        self.assertEqual(parsed_key, key)
+        self.assertEqual(parsed_value, value)
+
+    def test_GIVEN_an_empty_record_declaration_WHEN_parse_record_THEN_parser_can_extract_the_information(self):
+        rec_type, rec_name = "ai", "$(P)TEST"
+
+        # record(ai, "$(P)TEST") {}
+
+        lexer = MockLexer()\
+            .add_record_header(rec_type, rec_name)\
+            .add_token(TokenTypes.L_BRACE)\
+            .add_token(TokenTypes.R_BRACE)
+
+        parsed_record = Parser(lexer).record()
+
+        self.assertEqual(parsed_record["name"], rec_name)
+        self.assertEqual(parsed_record["type"], rec_type)
+        self.assertEqual(parsed_record["fields"], [])
+        self.assertEqual(parsed_record["infos"], [])
+        self.assertEqual(parsed_record["aliases"], [])
+
+    def test_GIVEN_an_empty_record_declaration_containing_a_field_and_an_info_WHEN_parse_record_THEN_parser_can_extract_the_information(self):
+        rec_type, rec_name = "ai", "$(P)TEST"
+        field_name, field_val = "PINI", "YES"
+        info_name, info_val = "alarm", "TEST_01"
+
+        # record(ai, "$(P)TEST") {
+        #     field(PINI, "YES")
+        #     info(alarm, "TEST_01")
+        # }
+        lexer = MockLexer()\
+            .add_record_header(rec_type, rec_name)\
+            .add_token(TokenTypes.L_BRACE)\
+            .add_field(field_name, field_val)\
+            .add_info_field(info_name, info_val)\
+            .add_token(TokenTypes.R_BRACE)
+
+        parsed_record = Parser(lexer).record()
+
+        self.assertEqual(parsed_record["name"], rec_name)
+        self.assertEqual(parsed_record["type"], rec_type)
+        self.assertEqual(parsed_record["fields"], [(field_name, field_val)])
+        self.assertEqual(parsed_record["infos"], [(info_name, info_val)])
+        self.assertEqual(parsed_record["aliases"], [])
+
+    def test_GIVEN_multiple_records_WHEN_parse_as_db_THEN_parser_can_extract_the_information(self):
+        rec_type_1 = "ai"
+        rec_type_2 = "bi"
+        rec_name_1 = "$(P)TEST1"
+        rec_name_2 = "$(P)TEST2"
+
+        # record(ai, "$(P)TEST1") {
+        # }
+        # record(bi, "$(P)TEST2") {
+        # }
+        lexer = MockLexer()\
+            .add_record_header(rec_type_1, rec_name_1)\
+            .add_token(TokenTypes.L_BRACE)\
+            .add_token(TokenTypes.R_BRACE)\
+            .add_record_header(rec_type_2, rec_name_2)\
+            .add_token(TokenTypes.L_BRACE)\
+            .add_token(TokenTypes.R_BRACE)
+
+        parsed_db = Parser(lexer).db()
+
+        self.assertEqual(len(parsed_db), 2)
+        rec1, rec2 = parsed_db
+
+        self.assertEqual(rec1["type"], rec_type_1)
+        self.assertEqual(rec2["type"], rec_type_2)
+        self.assertEqual(rec1["name"], rec_name_1)
+        self.assertEqual(rec2["name"], rec_name_2)
```

Comment:
add alias inside and outside of a record

Reply:
done